pkg/cloud/services/compute/instance.go

```diff
@@ -808,8 +817,9 @@ func (s *Service) GetInstance(resourceID string) (instance *infrav1.Instance, er
 	if resourceID == "" {
 		return nil, fmt.Errorf("resourceId should be specified to get detail")
 	}
+	mc := metrics.NewMetricPrometheusContext("server", "get")
 	server, err := servers.Get(s.computeClient, resourceID).Extract()
-	if err != nil {
+	if mc.ObserveRequest(err) != nil {
 		if capoerrors.IsNotFound(err) {
```

Comment:
same here and maybe also at som other `capoerrors.Is...`?

Reply:
I checked the other `capoerrors`. For `capoerrors.IsRetryable` I decided to keep it as is, to also get metrics for the retries.
xref: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/2af841db0cbcd3ce493526644438494bade7cd98/pkg/cloud/services/compute/instance.go#L664-L669